### PR TITLE
ADO 107231 identical ids

### PR DESCRIPTION
**Pull request by Maxim-Lam**
### Description

ids for the BenefitCards and overlapped with those of the "on this page" ListLinks.

Replaced ids for both with unique identifiable ids

List of proposed changes:

- replaced duplicate ids from benefitsCard and ListLinks to unique ones

### What to test for/How to test

- navigate to /eligibility
- fill out form with valid data
- inspect ID on the next page for the "on this page" links and the benefits cards "apply for [...]" and "learn more [...]"
- notice unique IDs

### CODE REVIEW

File: components/ResultsPage/index.tsx [hunk 1]
```diff
@@ -74,31 +74,42 @@ const ResultsPage: React.VFC<{
   let listLinks: {
     text: string
     url: string
+    idSuffix: string
   }[] = [
     {
       text: getEligibleLinkText(summary.entitlementSum, tsln),
       url: '#eligible',
+      idSuffix: 'eligilble',
```
Comment (MarcoGoC) [inline components/ResultsPage/index.tsx]: typo

File: components/ResultsPage/BenefitCard.tsx [hunk 2]
```diff
@@ -71,13 +77,13 @@ export const BenefitCard: React.VFC<{
...
         {links &&
-          links.map(({ text, url, icon, alt }, index) => (
+          links.map(({ text, url, icon, alt, idSuffix }, index) => (
             <div key={index} className="flex items-center py-4 text-content">
               <div>
                 <Image src={`/${icon}.png`} alt={alt} width="30" height="44" />
               </div>
               <div className="pl-5 w-full">
-                <DSLink id={`link${index}`} href={url} text={text} />
+                <DSLink id={`${idSuffix}Link${index}`} href={url} text={text} />
```
Comment (MarcoGoC) [inline components/ResultsPage/BenefitCard.tsx]: suggestion use Link${benefitKey}-${index}  instead, so idSuffix is not only not required, but no other change is required.

File: components/ResultsPage/index.tsx [hunk 3]
```diff
@@ -74,31 +74,42 @@ const ResultsPage: React.VFC<{
   let listLinks: {
     text: string
     url: string
+    idSuffix: string
```
Comment (MarcoGoC) [inline components/ResultsPage/index.tsx]: with the suggestion above this is not required

File: components/ResultsPage/BenefitCard.tsx (hunk 2 above)
Comment (Maxim-Lam) [reply components/ResultsPage/BenefitCard.tsx]: good idea

File: components/ResultsPage/index.tsx (hunk 3 above)
Comment (Maxim-Lam) [reply components/ResultsPage/index.tsx]: @MarcoGoC  An ID suffix is in someway required for listLinks as unlike the Benefits card, there is no benefitKey passed down to them. I did remove them in lieu of the benefits key on the BenefitsCard component.

**Review by MarcoGoC — APPROVED** [review]
Thanks Max

**Pull request merged**